evalml/utils/gen_utils.py

```diff
@@ -235,7 +235,10 @@ def get_importable_subclasses(base_class, used_in_automl=True):
         if "evalml.pipelines" not in cls.__module__:
             continue
         try:
-            cls()
+            if cls.name == "Time Series Regularizer":
+                cls(time_index="date")
```

Comment:
Added to address @chukarsten's comments above regarding making `time_index` a required param